docs/extended_api/synchronization_primitives/atomic_ref.md

```diff
@@ -0,0 +1,101 @@
+---
+grand_parent: Extended API
+parent: Synchronization Primitives
+nav_order: 1
+---
+
+# `cuda::atomic_ref`
+
+Defined in header `<cuda/atomic>`:
+
+```cuda
+template <typename T, cuda::thread_scope Scope = cuda::thread_scope_system>
+class cuda::atomic_ref;
+```
+
+The class template `cuda::atomic_ref` is an extended form of [`cuda::std::atomic_ref`]
+  that takes an additional [`cuda::thread_scope`] argument, defaulted to
+  `cuda::std::thread_scope_system`.
+It has the same interface and semantics as [`cuda::std::atomic_ref`], with the
+  following additional operations.
+This class additionally deviates from the standard by being backported to C++11.
+
+## Limitations
+
+`cuda::atomic_ref<T>` and `cuda::std::atomic_ref<T>` may only be instantiated with types sized between 4 and 8 bytes.
+
+Constructing multiple `cuda::std::atomic_ref<T, Sco>` with the same reference, but different values for `Sco` is illegal.
+
+`cuda::std::atomic_ref<T>` may be used with structures, however padded objects will result in undefined behaviour.
+
+## Atomic Extrema Operations
+
+| [`cuda::atomic_ref::fetch_min`] | Atomically find the minimum of the stored value and a provided value. `(member function)` |
+| [`cuda::atomic_ref::fetch_max`] | Atomically find the maximum of the stored value and a provided value. `(member function)` |
+
+## Concurrency Restrictions
+
+An object of type `cuda::atomic_ref` or [`cuda::std::atomic_ref`] shall not be accessed
+  concurrently by CPU and GPU threads unless:
+- it is in unified memory and the [`concurrentManagedAccess` property] is 1, or
+- it is in CPU memory and the [`hostNativeAtomicSupported` property] is 1.
```

Comment:
What problem does this solve? 

It sounds like this comes from the `atomic` specification, but that's there because the atomic operations are performed on the storage of the `atomic` type. That's not _always_ the case for `cuda::atomic_ref`, e.g., for all cases supported above `atomic_ref` is just a pointer, and loading it from any thread that can access that memory is fine. If that memory is going to be modified concurrently by another thread, then the load would need to be atomic, and then all wording for data-races applies.